sysl2/sysl/msg/Message.go

```diff
@@ -0,0 +1,166 @@
+package msg
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Msg holds message id and params to generate message string
+type Msg struct {
+	MessageID   int      `json:"message_id"`
+	MessageData []string `json:"message_data"`
+}
+
+// LogMessage holds message string and type of message
+type LogMessage struct {
+	Message string `json:"message"`
+	MsgType int    `json:"msg_type"`
+}
+
+const (
+	ERROR = 100
+	WARN  = 200
+	INFO  = 300
+	TITLE = 900
+	UNDEF = 500
+
+	ErrValidationFailed        = 400
+	ErrEntryPointUndefined     = 401
+	ErrInvalidEntryPointReturn = 402
+	ErrUndefinedView           = 403
+	ErrInvalidReturn           = 404
+	ErrMissingReqField         = 405
+	ErrExcessAttr              = 406
+	ErrInvalidOption           = 407
+	ErrInvalidUnary            = 408
+	ErrRedefined               = 409
+
+	WarnValidatedWithWarn = 300
+	WarnValidationSkipped = 301
+
+	InfoValidatedSuccessfully = 200
+	TitleViewName             = 999
+)
+
+// nolint:gochecknoglobals
+var (
+	Messages = map[int]string{
+		ErrValidationFailed:        "Validation failed",
+		ErrEntryPointUndefined:     "Entry point view: (%s) is undefined",
+		ErrInvalidEntryPointReturn: "Return type of entry point view: (%s) should be (%s)",
+		ErrUndefinedView:           "View (%s) is undefined",
+		ErrInvalidReturn:           "In view (%s), return type should be (%s)",
+		ErrMissingReqField:         "Missing required field (%s) in View (%s) with return Type (%s)",
+		ErrExcessAttr:              "Excess Attribute (%s), defined in View (%s), having return Type (%s)",
+		ErrInvalidOption:           "In View %s, (%s) does not match any of the options for return Type (%s)",
+		ErrInvalidUnary:            "In view (%s), unary operator used with invalid type: (%s)",
+		ErrRedefined:               "In view (%s), (%s) is already defined",
+
+		WarnValidatedWithWarn: "Validated with warnings",
+		WarnValidationSkipped: "Validation skipped. Reason: %s",
+
+		InfoValidatedSuccessfully: "Validation success",
+
+		TitleViewName: "Error in %s",
+	}
+
+	MessageType = map[int]int{
+		ErrValidationFailed:        ERROR,
+		ErrEntryPointUndefined:     ERROR,
+		ErrInvalidEntryPointReturn: ERROR,
+		ErrUndefinedView:           ERROR,
+		ErrInvalidReturn:           ERROR,
+		ErrMissingReqField:         ERROR,
+		ErrExcessAttr:              ERROR,
+		ErrInvalidOption:           ERROR,
+		ErrInvalidUnary:            ERROR,
+		ErrRedefined:               ERROR,
+
+		WarnValidatedWithWarn: WARN,
+		WarnValidationSkipped: WARN,
+
+		InfoValidatedSuccessfully: INFO,
+		TitleViewName:             TITLE,
+	}
+)
+
+func NewMsg(messageID int, messageData []string) *Msg {
+	return &Msg{MessageID: messageID, MessageData: messageData}
+}
+
+func (m Msg) String() string {
+	message, exists := Messages[m.MessageID]
+
+	if !exists {
+		fmt.Println("[validationError.String] invalid message ID")
```

Comment:
Don't print to stdout. Use logrus at least, but better to panic or return this message as the string value.